index.js

```diff
@@ -244,6 +246,83 @@ class GithubScm extends Scm {
             }
         };
     }
+
+    /**
+     * Get data for a specific repo
+     * @method getCommitSha
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUrl     The scmUrl to get data of
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoInfo(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'get',
+                token,
+                params: scmInfo
+            }, (error, repoInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                return resolve(repoInfo);
+            });
+        });
+    }
+
+    /**
+     * Get a url for a specific branch
+     * @method getRepoId
+     * @param  {Object}    config        Configuration
+     * @param  {String}    config.scmUrl The scmUrl to get url of
+     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getBranchUrl(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'getBranch',
+                token,
+                params: scmInfo
+            }, (error, branchInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                // eslint-disable-next-line no-underscore-dangle
+                return resolve(branchInfo._links.html);
+            });
+        });
+    }
+
+    /**
+     * Get repository object for storing new pipeline
+     * @method getRepoId
+     * @param {Object}    config        Configuration
+     * @param {String}    config.scmUrl The scmUrl to generate ID
+     * @param {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoId(config) {
```

Comment:
No worries @catto. I didn't know @stjohnjohnson suggested `getRepoId` should return this structure. I still think it's misleading though. Let's discuss with him about this. Will get back to you on Monday! Thanks for your good work! 😃